### ISSUE TITLE
Better Tire Support for Vehicle Mods

### ISSUE THREAD

**Issue by swampatha** [stblr/mkw-sp#124]
Some custom vehicle mods, such as many of TravixMan's MKDD and MKT ports, include custom BSP files to change the position of the tires. This works for offline play, but there are several issues with them:

- Most importantly, changing the BSP file changes the hitbox of the kart which makes the use of them online unsafe.
- The BSP files are loaded through the Common.szs which makes it harder for less-experienced members to use them (I see questions about them from time to time in support servers)
- The BSPs are universal for all vehicles on that slot, so if you have a custom kart mod over Mario on the Wild Wing, it will change the tire positions for all Wild Wings, even if they don't use that kart.
- Even with BSPs, it's not possible (to mine or Travix' knowledge) to rescale tires, meaning some ports from other games are not possible.

I don't know if it would be possible, but for those who like to use vehicle mods, it would be great to see some kind of file that can be included within the mod's SZS that would allow for visual-only tire changes. It could possibly be a slightly modified BSP file, some custom file type, or even just some JSON file (something similar to what MK8's parts.bin tool generates). This could allow for improved custom vehicles and the ability to use them in online play when that is added to MKW-SP.

**Comment by stblr** [comment]
This is about purely visual mods, correct? The scale is controlled by `kartPartsDispParam`/`bikePartsDispParam` AFAIK, although there is no character differentiation either. The file format can be something relatively simple that would override `Common.szs` files for graphical settings.

**Comment by swampatha** [comment]
Yeah, this would be for purely visual changes. I did not know about the kartPartsDispParam scaling settings though, so thank you.
